components/elective/ElectiveDetailsCard.tsx

```diff
@@ -0,0 +1,110 @@
+import MultilangText from "@/components/common/MultilingualText";
+import InformationCard from "@/components/lookup/people/InformationCard";
+import PeopleChipSet from "@/components/person/PeopleChipSet";
+import RoomChip from "@/components/room/RoomChip";
+import cn from "@/utils/helpers/cn";
+import getLocaleString from "@/utils/helpers/getLocaleString";
+import useLocale from "@/utils/helpers/useLocale";
+import { StylableFC } from "@/utils/types/common";
+import { ElectiveSubject } from "@/utils/types/elective";
+import { UserRole } from "@/utils/types/person";
+import { ChipSet, Text } from "@suankularb-components/react";
+import { useTranslation } from "next-i18next";
+
+/**
+ * A card similar to a Lookup Details Card that displays details of an Elective
+ * Subject.
+ *
+ * @todo Should accept `electiveSubject` prop.
+ */
+const ElectiveDetailsCard: StylableFC = ({ style, className }) => {
+  const locale = useLocale();
+  const { t } = useTranslation("elective", { keyPrefix: "detail" });
+
+  // Mock data
+  const electiveSubject = {
+    id: 1,
+    name: { th: "การสร้างเว็บเพจ 1", "en-US": "Webpage 1" },
+    code: { th: "ว20281", "en-US": "SC20281" },
+    description: {
+      th: "คอร์สที่ออกแบบมาเพื่อสอนการสร้างและออกแบบเว็บไซต์อย่างมีประสิทธิภาพ",
+      "en-US":
+        "A course designed to teach you how to create and design effective \
+        websites.",
+    },
+    room: "6306",
+    credit: 1.0,
+    teachers: [
+      {
+        id: "d84bd1a6-4659-496e-bde3-afdaeb1221c9",
```

Comment:
Is this the ID for teacher in supabase database?

Reply:
Yes, that is pretty obvious, no?

Reply:
**If you’re asking if this conforms to the ID system we use in Supabase:** yes. It’s UUID.
**If you’re asking if this is the ID for T. Wiyada:** yes. We needed to you their ID so `WithPersonDetails` can fetch their data from the database. That component was written before the API spec was even available, so we did it the old-fashioned way.
**If you’re asking if this is the ID code used by the school:** no. As far as I know, teachers don’t have an ID system outside of MySK.